addons/purchase/models/purchase.py

```diff
@@ -1058,6 +1058,10 @@ def _onchange_quantity(self):
                     self.order_id.company_id,
                     self.date_order or fields.Date.today(),
                 )
+
+            if self.product_uom:
```

Comment:
I'm ok with the patch. It would be really nice to have a little test with that. For instance adding a step in https://github.com/odoo/odoo/blob/70e57269535d3986df18f0ae4366aa3b80fee354/addons/purchase_stock/tests/test_onchange_product.py#L28
Thanks